sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/metadata/TableMetaDataLoader.java

```diff
@@ -161,7 +165,10 @@ private boolean isTableExist(final Connection connection, final String catalog,
                 String columnName = resultSet.getString(COLUMN_NAME);
                 String columnType = resultSet.getString(TYPE_NAME);
                 boolean isPrimaryKey = primaryKeys.contains(columnName);
-                Optional<ColumnMetaData> columnMetaData = getColumnMetaData(logicTableName, columnName, columnType, isPrimaryKey, generateKeyColumnName, encryptRule, derivedColumns);
+                boolean isNullable = isPrimaryKey || resultSet.getBoolean(IS_NULLABLE);
```

Comment:
'isNullable' means the column can be null. Why the primaryKey can be null?

Reply:
fix it.
`boolean isNotNull = isPrimaryKey || !resultSet.getBoolean(IS_NULLABLE);`